api/analyzedphenotypes.systemvars.api.inc

```diff
@@ -0,0 +1,206 @@
+      foreach($suffix as $term) {
+        $systemvars[$set]['ap_' . $term] = $var_prefix . $term;
+      }
+    }
+  }
+
+  return $systemvars;
+}
+
+
+///////
+
+
+/**
+ * CONTROL/SWITCHBOARD.
+ * Pass a request to a corresponding function.
+ * @TODO: leave out when api reorganize/update is complete and use functions below.
+ */
+function analyzedphenotypes_systemvariables($property = null, $parameter = null) {
+  $systemvars = null;
+
+  if (is_null($property)) {
+    // Default.
+    $property = 'default';
+  }
+
+  // Process request.
+  // Default to all system variables.
+  $command = array(
+    'default' => array(
+      'func'  => 'ap_define_variablenames',
+      'param' => null,
+      'opts'  => null,
+    ),
+    'options' => array(
+      'func'  => 'ap_get_systemvariable',
+      'param' => array('variablename' => 'allvars'),
+      'opts'  => array('set' => 'options'),
+    ),
+    'cvdbon' => array(
+      'func'  => 'ap_get_systemvariable',
+      'param' => array('variablename' => 'allvars'),
+      'opts'  => array('set' => 'cvdbon'),
+    ),
+    'terms' => array(
+      'func'  => 'ap_get_systemvariable',
+      'param' => array('variablename' => 'allvars'),
+      'opts'  => array('set' => 'terms'),
+    ),
+    'genus_vars' => array(
+      'func'  => 'ap_get_systemvariable',
+      'param' => array('variablename' => $parameter['genus']),
+      'opts'  => array('set' => 'cvdbon', 'suffix' => 'allsuffix'),
+    ),
+  );
+
+  // Function name, parameter (project name key or name), additional options.
+  return call_user_func($command[$property]['func'], $command[$property]['param'], $command[$property]['opts']);
+}
+
+
+///////
+
+
+/**
+ * FUNCTION:
+ * Fetch system variable name.
+ * Dependencies: ap_define_variablenames().
+ *
+ * @param $identifiers
+ *   An array with the following keys:
+ *     - variablename : term, option or genus keyword unique to a system variable.
+ * @param $options
+ *   An array to be used as additional options to be performed to returned result.
+ *     - set : the set a term belongs to. Default to CV, DB and ON (ontology) set.
+ *       allvars - all variables in a set.
+ *       terms - controlled vocabulary terms method, year etc.
+ *       cvdbon - for controlled vocabulary, database and ontology configuration.
+ *       options - module options such as allow new traits to be added.
+ *     - suffix : when requesting variables for a genus, this key holds value for cv, database or ontology settings.
+ *       allsuffix - request cv, db and ontology variable set for a genus.
+ *       cv - controlled vocabulary configuration only of a genus.
+ *       db - database configuration only  of a genus.
+ *       on - ontology configuration only of a genus.
+ *
+ * @return
+ *   A String, variable name or an array a set of variable names.
+ */
+function ap_get_systemvariable($identifiers, $options = array('set' => 'cvdbon', 'suffix' => 'cv')) {
```

Comment:
This function name is misleading. I expect to get the value of the variable, not the name... Perhaps `ap_get_variablenames()` would be better? This also makes the name work better with the first function since they act as a pair.